src/boot/z_std_dma.c

```diff
@@ -210,6 +210,9 @@ const char* DmaMgr_GetFileNameImpl(u32 vrom) {
     }
     //! @bug Since there is no return, in case the file isn't found, the return value will be a pointer to the end
     // of gDmaDataTable
+#ifdef AVOID_UB
+    return (const char*)iter;
```

Comment:
```suggestion
    return NULL;
```
we may as well have code that makes sense, since I wouldn't call casting iter to a string well-defined behavior

Reply:
Recasting a non-function pointer is pretty much always okay unless the alignment requirements are not the same: https://stackoverflow.com/questions/4810417/when-is-casting-between-pointer-types-not-undefined-behavior-in-c

*Dereferencing*, on the other hand...

Reply:
This is the only discussion left I'd like to see resolved, so the question is,
Does `return (const char*)iter;` lead to UB?

Reply:
Hmm, it will read whatever is after the end of the nonzero stuff in the dma table. But the iteration terminating relies on the next thing being zero anyway. So the function actually forces that the dma table end in a row of zeros to avoid UB in the first place, so it will just point to bytes which will be interpreted as `'\0'`. So provided that we end the dma table in a row of zeros, the behaviour is well-defined. (You can always dereference non-null pointers to read chars, according to the C standard, according to that link I posted earlier.)

Reply:
And it looks like we do not currently terminate the dma table with 0s, so there's UB elsewhere in this function already.

Reply:
Oh, yes we do, I didn't spot it in the rather Byzantine way it's set up now
```c

DmaEntry gDmaDataTable[] = {
#include "tables/dmadata_table.h"
    { 0 },
};```

Reply:
So if this list were to go all the way through without finding a match, it would end with `iter` pointing to this padding right? and would dereference to a string with only a null terminator? 
if so then I think we're fine

Reply:
it's sketch but I don't see a compiler doing something weird enough that it wouldn't be fine, so ok sure

but it's pretty weird to resort to "oh nice there's zeros here so ido's regalloc does something accidentally sensible" to justify this choice
the game does expect a zero vromEnd, but it's still reading vromStart as a string (the first (null) byte of it at least)

Reply:
Thinking about it, it's probably more appropriate to `return "";` to ensure it's always an empty string, preserving the behavior even more accurately.